e2e_playwright/st_experimental_audio_input_test.py

```diff
@@ -0,0 +1,384 @@
+    )
+
+
+def _test_download_audio_file(app: Page, locator: FrameLocator | Locator):
+    audio_input = locator.get_by_test_id("stAudioInput").nth(1)
+    audio_input.get_by_role("button", name="Record").click()
+    app.wait_for_timeout(1500)
+
+    stop_recording(audio_input, app)
+
+    with app.expect_download() as download_info:
+        download_button = audio_input.get_by_role("button", name="Download recording")
+        download_button.click()
+
+    download = download_info.value
+    file_name = download.suggested_filename
+
+    assert file_name == "recording.wav"
+
+
+@pytest.mark.only_browser("chromium")
+def test_audio_input_file_download(app: Page):
+    """Test that the audio input file can be downloaded."""
+    app.context.grant_permissions(["microphone"])
+
+    _test_download_audio_file(app, app.locator("body"))
+
+
+@pytest.mark.only_browser("chromium")
+def test_audio_input_file_download_in_iframe(iframed_app: IframedPage):
+    """Test that the audio input file can be downloaded within an iframe."""
+
+    page: Page = iframed_app.page
+    frame_locator: FrameLocator = iframed_app.open_app(None)
+
+    _test_download_audio_file(page, frame_locator)
+
+
+@pytest.mark.only_browser("chromium")
+def test_audio_input_callback(app: Page):
+    """Test that the callback is triggered when audio input changes."""
+    # Initial state before any interaction
+    expect(app.get_by_text("Audio Input Changed: False")).to_be_visible()
+
+    # Simulate recording interaction
+    audio_input = app.get_by_test_id("stAudioInput").nth(5)
+    audio_input.get_by_role("button", name="Record").click()
+    app.wait_for_timeout(1500)
+
+    stop_recording(audio_input, app)
+
+    ensure_waveform_rendered(audio_input)
+
+    # Verify the callback updated the UI
+    expect(app.get_by_text("Audio Input Changed: True")).to_be_visible()
+
+
+@pytest.mark.only_browser("chromium")
+def test_audio_input_remount_keep_value(app: Page):
+    """Test that the audio input component remounts without resetting its value."""
+    expect(app.get_by_text("audio_input-after-sleep: False")).to_be_visible()
+
+    # Simulate recording interaction
+    audio_input = app.get_by_test_id("stAudioInput").nth(6)
+    audio_input.scroll_into_view_if_needed()
+    audio_input.get_by_role("button", name="Record").click()
+    app.wait_for_timeout(1500)
+
+    stop_recording(audio_input, app)
+
+    wait_for_app_run(app)
+
+    # Ensure the value is retained after remount
+    expect(app.get_by_text("audio_input-after-sleep: True")).to_be_visible()
+
+    # Unmount the component and verify the value is still retained
+    click_button(app, "Create some elements to unmount component")
+    expect(app.get_by_text("audio_input-after-sleep: True")).to_be_visible()
+
+    ensure_waveform_rendered(audio_input)
+
+
+@pytest.mark.only_browser("chromium")
+def test_audio_input_works_in_forms(app: Page):
```

Comment:
This test seems to be flaky at the moment:

<img width="757" alt="image" src="https://github.com/user-attachments/assets/90fd289b-f9b0-40e8-b074-b1f1ae089ee1">

Before adding this for experimental_audio_input and audio_input it might be good to look into it if we can fix it.